v2/tools/generator/internal/astbuilder/declarations.go

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astbuilder
+
+import (
+	"fmt"
+
+	"github.com/dave/dst"
+)
+
+func Declarations(declarations ...any) []dst.Decl {
```

Comment:
I'm not sold that the flexibility of accepting `any` here is worth loss of typing. It reads well once you've written it, but I suspect to actually write a call to this you'll need to come look at the impl to understand what it accepts. A doc string here might help, but wondering if something like this would be better:

```
func Declarations(declarations ...dst.Decl) []dst.Decl

// TODO: Name isn't great...
func UnionDeclarations(declarations ...[]dst.Decl) []dst.Decl
```

Whose usage is:
```
astbuilder.UnionDeclarations(
		astbuilder.Declarations(declaration),
		interfaceDeclarations,
		methodDeclarations)
```

Though I suppose you could make the argument that `Declarations(declarations ...dst.Decl) []dst.Decl` isn't much better than just making the slice inline, which could produce this instead:
```
astbuilder.Declarations(
		[]dst.Decl{declaration},
		interfaceDeclarations,
		methodDeclarations)
```

My 2c is that either of the above are still pretty readable (maybe the second one wins?), with the advantage of typehinting/intellisense for them actually being useful in a way that the `any` parameter prevents.